### PR TITLE
Contract init with args

### PR DESCRIPTION
**Pull request by kpob**
* Update macro code
* Update unit tests

### CODE REVIEW

File: macros/src/contract.rs [hunk 1]
```diff
@@ -113,9 +117,11 @@ pub mod interface {
...
 
 pub mod utils {
-    use proc_macro2::{Ident, TokenStream};
-    use quote::{quote, TokenStreamExt};
-    use syn::{FnArg, Pat, TraitItemMethod, Type, TypePath};
+    use proc_macro2::{Ident, Span, TokenStream};
+    use quote::{format_ident, quote, TokenStreamExt};
+    use syn::{punctuated::Punctuated, token, FnArg, Pat, Token, TraitItemMethod, Type, TypePath};
+
+    use crate::parser::ContractTrait;
```
Comment (zie1ony) [inline macros/src/contract.rs:124]: I don't think it's a good idea to name a struct `ContractTrait`. Maybe rename it to `ContractDef`?

File: macros/src/contract.rs [hunk 2]
```diff
@@ -180,4 +186,36 @@ pub mod utils {
             }
         }
     }
+
+    pub fn find_method<'a>(
+        input: &'a ContractTrait,
+        method_name: &'static str,
```
Comment (zie1ony) [inline macros/src/contract.rs:192]: I checked and it is possible to remove `static`, so it looks like: `method_name: &str,`.

Comment (zie1ony) [inline macros/src/contract.rs:192]: I checked and it is possible to remove `static`, so it looks like: `method_name: &str,`.

**Review by zie1ony — APPROVED** [review]
Small rework required.

File: macros/src/contract.rs (hunk 1 above)
Comment (kpob) [reply macros/src/contract.rs:124]: Changed already in #36

File: macros/src/contract.rs (hunk 2 above)
Comment (kpob) [reply macros/src/contract.rs]: Done

**Pull request merged**